go/vt/vtgate/planbuilder/operators/cte_merging.go

```diff
@@ -79,17 +79,21 @@ func mergeCTE(ctx *plancontext.PlanningContext, seed, term *Route, r Routing, in
 	hz := in.Horizon
 	hz.Source = term.Source
 	newTerm, _ := expandHorizon(ctx, hz)
+	cte := &RecurseCTE{
+		binaryOperator: binaryOperator{
+			LHS: seed.Source,
+			RHS: newTerm,
+		},
+		Predicates: in.Predicates,
+		Def:        in.Def,
+		LeftID:     in.LeftID,
+		OuterID:    in.OuterID,
+		Distinct:   in.Distinct,
+	}
 	return &Route{
 		Routing: r,
-		Source: &RecurseCTE{
-			Predicates: in.Predicates,
-			Def:        in.Def,
-			Seed:       seed.Source,
-			Term:       newTerm,
-			LeftID:     in.LeftID,
-			OuterID:    in.OuterID,
-			Distinct:   in.Distinct,
-		},
+		unaryOperator: unaryOperator{
+			Source: cte},
```

Comment:
we should almost have a `newUnaryOperator(x)` and `newBinaryOperator(x, y)` function